### PR TITLE
chore: Update Dockerfile to node:18

### PR DESCRIPTION
**Pull request by landrito**
Addresses #2514

**Summary**

Upgrades base image in Dockerfile to `node:18.20.1` to resolve vulnerabilities flagged on the docker image.

**Checklist**

- The basics
  - [x] I tested these changes manually in my local or dev environment
- Tests
  - [ ] Added or updated
  - [x] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [x] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [x] N/A

**Screenshots**

N/A

**Additional context**

N/A

### CODE REVIEW

**Comment by landrito** [comment]
@kaylachun Friendly ping on this CL. Thanks!

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:16 as compiler
+FROM node:18.20.1 as compiler
```
Comment (daniel-white) [inline Dockerfile:1]: i dont think we want to be this specific - `node:18` should be sufficent

**Review by kaylachun — APPROVED** [review]

**Comment by landrito** [comment]
Hi @kaylachun,

Our team is attempting to release a feature that is dependent on the Prism Docker image in Docker Hub. However, we were unable to obtain approval from our security team due to the unresolved vulnerabilities on the docker image. With this merge, could you cut a new version and release to docker hub in the near future? 

Thank you for your help!

**Comment by landrito** [comment]
Thanks!!

**Pull request merged**